specs/src/specs/header-sync.md

```diff
@@ -0,0 +1,57 @@
+# P2P Header Sync
+
+## Abstract
+
+The nodes in the P2P network sync headers using the header sync service that implements the [go-header][go-header] interface. The header sync service consists of several components as listed below.
+
+| Component  | Description                                                                                                                                                                |
+| --- | --- |
+| store      | a `headerEx` prefixed [datastore][datastore] where synced headers are stored                                                                                                          |
+| subscriber | a [libp2p][libp2p] node pubsub subscriber                                                                                                                                            |
+| p2p server | a server for handling header requests between peers in the p2p network                                                                                                     |
+| exchange   | a client that enables sending in/out-bound header requests from/to the p2p network                                                                                         |
+| syncer     | a service for efficient synchronization for headers. When a p2p node falls behind and wants to catch up to the latest network head via p2p network, it can use the syncer. |
```

Comment:
You don't need all the extra spaces, the tables auto size to the largest row. I'd remove the spaces.

https://www.markdownguide.org/extended-syntax/#tables